# Tools employed for the African catfish genome project

## Raw data processing
- fastp 0.23.2 (filtering/trimming Illumina PE150 reads) |[Paper](https://doi.org/10.1093/bioinformatics/bty560) | [Github](https://github.com/OpenGene/fastp)

## K-mer size estimation

- Jellyfish 2.3.0 | [Paper](https://pubmed.ncbi.nlm.nih.gov/21217122/) | [Github](https://github.com/gmarcais/Jellyfish/tree/develop/swig)
- GenomeScope 2.0 | [Paper](https://www.nature.com/articles/s41467-020-14998-3) | [Github](https://github.com/tbenavi1/genomescope2.0)
- KAT 2.4.2 | [Paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5408915/) | [Github](https://github.com/TGAC/KAT)

## Assembly pipeline

- HiFiasm 0.16.1 | [Paper](https://doi.org/10.1038/s41592-020-01056-5) | [Github](https://github.com/chhylp123/hifiasm)

## Assembly scaffolding

## Assembly validation

- Merqury | [Paper](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-02134-9) | [Github](https://github.com/marbl/merqury)
- BUSCO | [Paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8476166/)

## Repeats annotation

## De novo transcripts assembly

## Genome annotation pipeline

## Orthology analysis


## Phylogenetics analysis
